pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -1750,3 +1756,22 @@ type AuthorizationValidationError struct {
 func (e *AuthorizationValidationError) Error() string {
 	return e.err
 }
+
+type ScrapeClass struct {
+	// Name of the scrape class.
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Name string `json:"name"`
+
+	// Default designates a scrape class as the default class.
+	// +optional
+	Default *bool `json:"default,omitempty"`
+
+	// TLSConfig section for scrapes.
+	// +optional
+	TLSConfig *TLSConfig `json:"tlsConfig,omitempty"`
+
+	// Authorization section for scrapes.
+	// +optional
+	Authorization *Authorization `json:"authorization,omitempty"`
```

Comment:
To keep it simple, I'd implement TLS configuration only (esp. because it was the main use case driving this request).